avm/res/network/nat-gateway/main.bicep

```diff
@@ -4,14 +4,14 @@ metadata description = 'This module deploys a NAT Gateway.'
 @description('Required. Name of the Azure Bastion resource.')
 param name string
 
-@description('Required. A list of availability zones denoting the zone in which Nat Gateway should be deployed.')
+@description('Required. If set to 1, 2 or 3, the availability zone is hardcoded to that value. If set to -1, no zone is defined. Note that the availability zone number here are the logical availability zone in your Azure subscription. Different subscriptions might have a different mapping of the physical zone and logical zone. To understand more, please refer to [Physical and logical availability zones](https://learn.microsoft.com/en-us/azure/reliability/availability-zones-overview?tabs=azure-cli#physical-and-logical-availability-zones) and [Distribute VMs and disks across availability zones](https://learn.microsoft.com/en-us/azure/virtual-machines/disks-high-availability#distribute-vms-and-disks-across-availability-zones).')
```

Comment:
Is "Distribute VMs and disks across availability zones" a useful link here? or is this the agreed description no matter the resource type?

Reply:
Fair. This stems from the spec. will fix it here and there

Reply:
Ref: https://github.com/Azure/Azure-Verified-Modules/pull/1939

Reply:
Approved both